Docs for notebook expressions

mazameli opened the pull request:
Because there was so much content for these new features, I elected to create a new page just for expressions, with links to it from the TOC and the main notebook doc page.

To-do
- [x] Add screenshots

mazameli committed:
start the ball rolling

mazameli committed:
add most of the content and the big table

mazameli committed:
add TOC link to new page

dacort commented on docs/users-guide/expressions.md:
```diff
@@ -0,0 +1,102 @@
+
+### Database limitations
+
+Certain database types don't support some of the above functions:
```
1. Is this an exhaustive list? 
2. How does this impact 3rd-party drivers? I know we're planning to update [the wiki](https://github.com/metabase/metabase/wiki/What's-new-in-0.35.0-for-Metabase-driver-authors) so we may want to comment here and link out to that.

mazameli replied:
1. This is going off of the spreadsheet that @sbelak provided, so it's exhaustive as far as I know.
2. Cool, good idea. I'll add a link to the wiki.

dacort replied:
Checked out the spreadsheet and, yes, it's exhaustive. 🙌 We should def make that public as it's a great reference.

dacort commented on docs/users-guide/expressions.md:
```diff
@@ -0,0 +1,102 @@
+- sqrt
+- variance
+
+Additionally, **Vertica**, **BigQuery**, and **Presto** only provide _approximate_ results for median and percentile.
```
Do we have specific references we can link out to here? For example, I see that BigQuery has both [`PERCENTILE_CONT`](https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#percentile_cont) and [`APPROX_QUANTILES`](https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#approx_quantiles). So does this mean that we're using the latter? The former doesn't appear to be approximate.

mazameli replied:
🤷‍♂ paging @sbelak

dacort replied:
OK, I see we're using `APPROX_QUANTILES` - Will comment in the other ticket to see what the reasoning is there and we can leave this as is for now.

dacort commented on docs/users-guide/expressions.md:
```diff
@@ -0,0 +1,102 @@
+In each of these three places, you can:
+
+- Use parentheses to group parts of your expression.
+- Use basic mathematical operators: `+`, `-`, `*` (multiply), `/` (divide).
```
I think it would be good to add some commentary one what types of fields you can use mathematical operators with. e.g. Can I subtract a bigint field from a string field? What happens when I do? Can I do math on timestamp fields? (Doesn't seem like with current `HEAD`.)

dacort reviewed (commented):
A few minor questions/comments. Otherwise looks good to me!

mazameli committed:
address dacort's feedback and add screenshots

mazameli commented:
Addressed @dacort's feedback and added screenshots, so I think all this could use is a glance from @sbelak if possible to answer Damon's questions about db-specific limitations (I just referred to that spreadsheet Simon linked to in the [math PR](https://github.com/metabase/metabase/pull/12096)).

dacort commented:
Sweet! I think the spreadsheet answered my questions so no other changes for the docs unless something changes in there. It would be great to publish the spreadsheet as a reference somewhere, but we can do that in another PR if necessary.

mazameli commented:
@dacort okay, cool. In that case I'm going to remove @sbelak as a reviewer. Mind giving me a green checkmark, Damon?
